packages/test-utils/lib/e2e/login.ts

```diff
@@ -18,32 +20,41 @@ export async function login(page: Page, phone?: string, text_username?: string,
   const now = DateTime.now();
   const authLoginContinueButton = page.locator('button:has-text("Continue")');
   await authLoginContinueButton.click();
-  let n = 0;
-  let code: string | undefined = '';
-  while (n < 3) {
-    n++;
-    await page.waitForTimeout(10000);
-    code = await getCode(now, text_username, text_password);
-    if (code) break;
-  }
+  const latestCode = await getLatestCode(now, text_username, text_password);
 
-  if (!code) {
-    throw new Error("Couldn't receive verification code with sms. Exceeded retries count");
+  if (!latestCode) {
+    throw new Error('No code received from SMS service');
   }
 
+  console.log(`Trying code: ${latestCode}`);
+
   const codeInput = page.locator('#code');
-  await codeInput.fill(code);
+  await codeInput.clear();
+  await codeInput.fill(latestCode);
+
   const verifyButton = page.locator('button:has-text("Continue")');
   await verifyButton.click();
-  await page.waitForTimeout(5000);
-  expect(await page.getByText('Code is invalid').isVisible()).toBeFalsy();
 
-  if (await page.getByText('Authorize App').isVisible()) {
-    const acceptButton = page.locator('button:has-text("Accept")');
+  expect(await page.getByText('Code is invalid').isVisible({ timeout: 5000 })).toBeFalsy();
+
+  try {
+    // optional thirtd party modal which may appear, it's tricky to handle, because text may be different, currently we just click on accept button, but maybe we can find a better way to handle it
```

Comment:
```suggestion
    // optional third party modal which may appear, it's tricky to handle, because text may be different, currently we just click on accept button, but maybe we can find a better way to handle it
```